store/src/java/com/zimbra/util/ExternalVolumeInfoHandler.java

```diff
@@ -238,4 +300,58 @@ public Boolean isVolumePresentInJson(int volumeId) throws ServiceException, JSON
         }
         return false;
     }
+
+    /**
+     * Write server json file
+     * @param data, dirPath, filePath
+     * @returns
+     * @throws
+     */
+    private void writeServerJSONFile(String data, String dirPath, String filePath) {
+        try {
+            File serverJSONBackupDir = new File(dirPath);
+            File serverJSONBackupFile = new File(filePath);
+            // check if dir exists
+            if (!serverJSONBackupDir.exists()) {
+                serverJSONBackupDir.mkdirs();
+            }
+            // check if file exists
+            if (!serverJSONBackupFile.exists()) {
+                serverJSONBackupFile.createNewFile();
+            }
+            Writer fileWriter = new FileWriter(serverJSONBackupFile, false); //overwrites file
+            fileWriter.write(data);
+            fileWriter.close();
+        } catch (IOException e) {
+            ZimbraLog.misc.error("Failure while writing Server JSON Backup File : ", e);
+        }
+    }
+
+    /**
+     * Read server json file
+     * @param filePath
+     * @returns String
+     * @throws
+     */
+    private String readServerJSONFile(String filePath) {
+        String result = null;
+        try {
+            File serverJSONBackupFile = new File(filePath);
+            // check if file exists
+            if (serverJSONBackupFile.exists()) {
+                BufferedReader reader = new BufferedReader(new FileReader(filePath));
+                String line = null;
+                while ((line = reader.readLine()) != null) {
+                    if (line.length() > 0) {
+                        result = line;
```

Comment:
Do you want to read a single line only ? or am I missing something?

Reply:
Yes I want to read single line only, as we don't have **'\n'** in json when we read single line only, all content/data gets read.
I have tested this

Reply:
Don't do this like, read the whole file. you never know

Reply:
Okay changed this logic.